### DISCUSSION TITLE
Errors shown in phpsysinfo

### DISCUSSION THREAD

**Discussion by Datore7** [mcguirepr89/BirdNET-Pi#830] [General]
Hello!

After upgrading BirdNET-Pi today I got for the first time ever errors when running phpsysinfo:

This appears when I press the warning sign:

/usr/local/bin/extra_info.sh - /usr/local/bin/extra_info.sh: command substitution: line 8: syntax error near unexpected token `!'
/usr/local/bin/extra_info.sh: command substitution: line 8: `dpkg -l | ! grep -q &quot;vcgencmd&quot;)&quot;'
Return value: 0

./xml.php on line 39
./includes/output/class.WebpageXML.inc.php on line 254 in function getJsonString()
./includes/output/class.WebpageXML.inc.php on line 241 in function getXMLString()
./includes/xml/class.XML.inc.php on line 914 in function getXml()
./includes/xml/class.XML.inc.php on line 903 in function _buildXml()
./includes/xml/class.XML.inc.php on line 941 in function _buildPlugins()
./plugins/viewer/class.viewer.inc.php on line 63 in function execute()
./includes/class.CommonFunctions.inc.php on line 382 in function executeProgram( "extra_info.sh", "", "........................................IPs.....................................\nLAN IP: 192.168.0.6\nPublic IP: 95.192.161.214\n.....................................Caddyfile..................................\nhttp:// https://birdpifyren.ddns.net {\n\troot * /home/birder/BirdSongs/Extracted\n\tfile_server browse\n\thandle /By_Date/* {\n\t\tfile_server browse\n\t}\n\thandle /Charts/* {\n\t\tfile_server browse\n\t}\n\tbasicauth /views.php?view=File* {\n\t\tbirdnet $2a$14$08rM8E0GzxexmZAsq5tuTejdmR1gwmesxfjv.AVZDaD7e6L9k.dPu\n\t}\n\tbasicauth /Processed* {\n\t\tbirdnet $2a$14$08rM8E0GzxexmZAsq5tuTejdmR1gwmesxfjv.AVZDaD7e6L9k.dPu\n\t}\n\tbasicauth /scripts* {\n\t\tbirdnet $2a$14$08rM8E0GzxexmZAsq5tuTejdmR1gwmesxfjv.AVZDaD7e6L9k.dPu\n\t}\n\tbasicauth /stream {\n\t\tejder $2a$14$zqql0oncQ.vrl5o3VP4F3uarDPkhBSa053K2tnt/B7bx9pn8X68lC\n\t}\n\tbasicauth /phpsysinfo* {\n\t\tbirdnet $2a$14$08rM8E0GzxexmZAsq5tuTejdmR1gwmesxfjv.AVZDaD7e6L9k.dPu\n\t}\n\tbasicauth /terminal* {\n\t\tbirdnet $2a$14$08rM8E0GzxexmZAsq5tuTejdmR1gwmesxfjv.AVZDaD7e6L9k.dPu\n\t}\n\treverse_proxy /stream localhost:8000\n\tphp_fastcgi unix//run/php/php7.4-fpm.sock\n\treverse_proxy /log* localhost:8080\n\treverse_proxy /stats* localhost:8501\n\treverse_proxy /terminal* localhost:8888\n\tlog {\n\toutput file /var/log/caddy/birder.log\n\t}\n}\n.................................... Crontab....................................\n\nSHELL=/bin/sh\nPATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin\n\n17 *\t* * *\troot cd / && run-parts --report /etc/cron.hourly\n25 6\t* * *\troot\ttest -x /usr/sbin/anacron || ( cd / && run-parts --report /etc/cron.daily )\n47 6\t* * 7\troot\ttest -x /usr/sbin/anacron || ( cd / && run-parts --report /etc/cron.weekly )\n52 6\t1 * *\troot\ttest -x /usr/sbin/anacron || ( cd / && run-parts --report /etc/cron.monthly )\n*/5 * * * * birder /usr/local/bin/disk_check.sh >/dev/null 2>&1\n*/3 * * * * birder /usr/local/bin/cleanup.sh >/dev/null 2>&1\n@reboot birder /usr/local/bin/cleanup.sh >/dev/null 2>&1\n0 1 * * 1 birder /usr/local/bin/weekly_report.sh >/dev/null 2>&1" )

I run 2 BirdNET-Pi, same error on both.

But everything seems to work fine.

Happy Easter everyone!

**Comment by ehpersonal38** [comment]
Thanks for the bug report! I've fixed this now.

**Reply by Datore7** [reply]
I tried to update now, but got this message for the first time ever:

(sh: 1: update_birdnet.sh: Permission denied)

**Comment by Datore7** [comment]
2 of all .sh files in scripts that had wrong rights:
print_diagnostic_info.sh
and
update_birdnet.sh
I chmoded them to 777 and now update seems to work.

**Comment by Datore7** [comment]
Now after the update the files:
print_diagnostic_info.sh
update_birdnet.sh
have this: -rw-rw-rw-

**Reply by ehpersonal38** [reply]
Thanks, I think I fixed this now. I wish Github showed when file perms were being changed in PRs.

**Reply by Datore7** [reply]
I had to chmod 777 update_birdnet.sh to be able to run it, after upgrade all sh files have "x" again, and all seems ok.
